### ISSUE TITLE
Error: 仅完成部分的 ReadProcessMemory 或 WriteProcessMemory 请求。 (os error 299)

### ISSUE THREAD

**Issue by lihailong00** [benfred/py-spy#728]
My Python version is 3.12.7
I use Windows 11
I run a python program in pycharm:

```python
import os
import time


def main():
    print(os.getpid())
    print(1)
    time.sleep(100)
    print(2)
    time.sleep(200)
    print(3)


if __name__ == '__main__':
    main()
```
then print the process id: 58396

then I input "py-spy top --pid 58396"
then Error occured:
```
Error: 仅完成部分的 ReadProcessMemory 或 WriteProcessMemory 请求。 (os error 299)
Reason: 仅完成部分的 ReadProcessMemory 或 WriteProcessMemory 请求。 (os error 299)
```

![image](https://github.com/user-attachments/assets/26a38c04-582a-4109-b744-dbdc9c85d4bc)

What's the promblems.

Thanks for your reply~

**Comment by tim-stephenson** [comment]
I am getting the same error:
```
Error: Only part of a ReadProcessMemory or WriteProcessMemory request was completed. (os error 299)
Reason: Only part of a ReadProcessMemory or WriteProcessMemory request was completed. (os error 299)
```
python 3.12
Windows 10
py-spy 0.4.0

**Comment by lihailong00** [comment]
Here's my code， function `find_urls_from_text` is stuck in an endless loop. I use `os.getpid()` to get the current process number. Then I execute `py-spy record -o profile.svg --pid ` in my terminal. then I am getting the error:
```
Error: Only part of a ReadProcessMemory or WriteProcessMemory request was completed. (os error 299)
Reason: Only part of a ReadProcessMemory or WriteProcessMemory request was completed. (os error 299)
```

```python
import os
import re
from typing import List


def find_urls_from_text(text: str) -> List[str]:
    """
    get websites from urls
    :param text:
    :return:
    """
    url_pattern = (
        r'(?i)\b((?:https?://|www\d{0,3}[.]|[a-z0-9.\-]+[.][a-z]{2,4}/)(?:[^\s()<>]+|\(([^\s()<>]+|(\([^\s()<>]+\)))*\))+(?:\(([^\s()<>]+|(\([^\s()<>]+\)))*\)|[^\s`!()\[\]{};:\'".,<>?«»“”‘’]))'
    )

    # 查找所有匹配项
    urls = re.findall(url_pattern, text, re.IGNORECASE)

    # 提取完整的网址
    cleaned_urls = [''.join(match[0].split()) for match in urls]

    return cleaned_urls


print(os.getpid())
s = '中国政府采购网(http://www.ccgp.gov.cn)、广西壮族自治区政府采购网(http://zfcg.gxzf.gov.cn)、广西壮族自治区招标投标公共服务平台（http://zbtb.gxi.gov.cn:9000/）、全国公共资源交易平台（广西·百色）+(http://ggzy.jg+swj.gxzf.gov.cn/bsggzy/）、广西百色市住房和城乡建设局（http://zjj.baise.gov.cn）上发布'
res = find_urls_from_text(s)
print(res)

```

**Comment by v01d-gh** [comment]
Same here. It's because of Python version. Py-spy doesn't support python 3.12 or 3.13 for sure yet. I can confirm that it works for Python 3.5. Also you can see Python 3.6 example in README.md file. But I don't know what is the maximum python version supported.

**Comment by tim-stephenson** [comment]
I agree the reason for the issue is python 3.12. I have been able to use py-spy fine on Windows with python 3.10. However, py-spy seems to be currently claiming to have python 3.12 and 3.13 support (See 0.4.0 release notes https://github.com/benfred/py-spy/releases/tag/v0.4.0). Such support many be Linux / MacOS only. If that is the case it should be clearly stated to users.

**Comment by akhramov** [comment]
@benfred hey, I had a debugging session today. Something is off with the GIL retrieval routines. I suspect it is something related to atomic pointers, or rather the way we read them. addr always ends up being `0x100000000`

```patch
diff --git a/src/stack_trace.rs b/src/stack_trace.rs
index cbfc51d..e271968 100644
--- a/src/stack_trace.rs
+++ b/src/stack_trace.rs
@@ -291,7 +291,7 @@ pub fn get_gil_threadid<I: InterpreterState, P: ProcessMemory>(
         let addr: usize = process.copy_struct(threadstate_address)?;
 
         // if the addr is 0, no thread is currently holding the GIL
-        if addr != 0 {
+        if addr != 0 && addr != 0x100000000 {
             let threadstate: I::ThreadState = process.copy_struct(addr)?;
             return Ok(threadstate.thread_id());
         }
````

**Comment by skrobchik** [comment]
With Python 3.13 I was getting the same error on Windows.

Downgraded to 3.11 and it worked.

**Comment by StevenChen16** [comment]
py-spy  currently claims to have Python 3.12 and 3.13 support, while I still face the same problem under py 3.12.8

**Comment by zengruizhao** [comment]
keep watching

**Comment by benfred** [comment]
This shoudl be fixed by https://github.com/benfred/py-spy/pull/774

**Comment by laundmo** [comment]
You can install the unreleased changes without having to compile yourself this way:
1. go to https://github.com/benfred/py-spy/actions/workflows/build.yml
2. click on the first run which lists `master`
3. at the left side, click on `build (windows-latest)`
4. find `upload windows wheels` (15th place) and expand it
5. the last output should be `Artifact download URL: ` - click on this

The downloaded zip file contains a wheel which you can `pip install ./py_spy[...].whl`

I didn't paste the URL here as these files might be temporary.

Maybe these files could be made more accessible, for example through a github pre-release?

**Comment by joseph-sch** [comment]
> This shoudl be fixed by [#774](https://github.com/benfred/py-spy/pull/774)

The issue is resolved for me in the latest build. If confirmed by others, **wouldn't it be worth a release?**

_My configuration:_

* Python: **3.12.9** (Anaconda)
* OS: Microsoft Windows 11 Pro 24H2 26100.4652 64-bit
* `py-spy` build: [wheels-windows.zip](https://productionresultssa14.blob.core.windows.net/actions-results/dd164be8-d869-4402-a40b-30b77df601cb/workflow-job-run-63bd4b1c-62d6-5185-baa8-a0c93d846fb4/artifacts/81d765cdce25c2dc02d0fbc9eddb540988f0ceb151736fc92e247f5b007b7b14.zip?rscd=attachment%3B+filename%3D%22wheels-windows.zip%22&se=2025-07-23T11%3A34%3A25Z&sig=jvEPlgmdVZDY7cfg89lZxvoB6mIlpgt1Eysw%2FWShAOY%3D&ske=2025-07-23T21%3A42%3A03Z&skoid=ca7593d4-ee42-46cd-af88-8b886a2f84eb&sks=b&skt=2025-07-23T09%3A42%3A03Z&sktid=398a6654-997b-47e9-b12b-9515b896b4de&skv=2025-05-05&sp=r&spr=https&sr=b&st=2025-07-23T11%3A24%3A20Z&sv=2025-05-05) from job [build (windows-latest)](https://github.com/benfred/py-spy/actions/runs/16324337720/job/46111911107) in [action 16324337720](https://github.com/benfred/py-spy/actions/runs/16324337720) at commit [405b56a](https://github.com/benfred/py-spy/commit/405b56a513e169fc0fb45ec42f610ce8a7eee53e)
